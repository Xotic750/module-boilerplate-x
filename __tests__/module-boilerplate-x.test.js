import testSubject from '../src/module-boilerplate-x';

describe('testSubject', function() {
  it('is a function', function() {
    expect.assertions(1);
    expect(typeof testSubject).toBe('function');
  });

  it('should throw when target not undefined', function() {
    expect.assertions(1);
    expect(function() {
      testSubject(null);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should return an object when target undefined', function() {
    expect.assertions(2);
    expect(typeof testSubject()).toBe('object');
    expect(testSubject()).not.toBeNull();
  });

  it('returned object should have a reserved property of true, IE<9', function() {
    expect.assertions(4);
    const obj = testSubject();
    expect(obj.delete).toBe(true);
    expect(obj.throws).toBe(true);
    expect(obj.blah).toBe(true);
    expect(obj.foo).toBeUndefined();
  });
});
