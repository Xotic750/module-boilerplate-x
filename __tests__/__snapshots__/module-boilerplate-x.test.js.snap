// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testSubject should throw when target not undefined 1`] = `"target is not undefined"`;
